test/integration/crud/misc_cursors.test.js

```diff
@@ -271,79 +210,47 @@ describe('Cursor', function () {
+      const collection = await db.createCollection('test_count.ext');
 
-          collection.find().count(err => {
-            expect(err).to.not.exist;
+      await collection.find().count();
 
-            function insert(callback) {
-              var total = 10;
+      async function insert() {
+        for (var i = 0; i < 10; i++) {
+          await collection.insert({ x: i }, configuration.writeConcernMax());
+        }
+      }
 
-              for (var i = 0; i < 10; i++) {
-                collection.insert({ x: i }, configuration.writeConcernMax(), e => {
-                  expect(e).to.not.exist;
-                  total = total - 1;
-                  if (total === 0) callback();
-                });
-              }
-            }
+      async function finished() {
+        let count = await collection.find().count();
+        test.equal(10, count);
+        test.ok(count.constructor === Number);
 
-            function finished() {
-              collection.find().count((err, count) => {
-                expect(err).to.not.exist;
-                test.equal(10, count);
-                test.ok(count.constructor === Number);
-
-                collection.find({}, { limit: 5 }).count((err, count) => {
-                  expect(err).to.not.exist;
-                  test.equal(5, count);
-
-                  collection.find({}, { skip: 5 }).count((err, count) => {
-                    expect(err).to.not.exist;
-                    test.equal(5, count);
-
-                    db.collection('acollectionthatdoesn').count((err, count) => {
-                      expect(err).to.not.exist;
-                      test.equal(0, count);
-
-                      var cursor = collection.find();
-                      cursor.count((err, count) => {
-                        expect(err).to.not.exist;
-                        test.equal(10, count);
-
-                        cursor.forEach(
-                          () => {},
-                          err => {
-                            expect(err).to.not.exist;
-                            cursor.count((err, count2) => {
-                              expect(err).to.not.exist;
-                              expect(count2).to.equal(10);
-                              expect(count2).to.equal(count);
-                              done();
-                            });
-                          }
-                        );
-                      });
-                    });
-                  });
-                });
-              });
-            }
+        count = await collection.find({}, { limit: 5 }).count();
+        test.equal(5, count);
 
-            insert(function () {
-              finished();
-            });
-          });
-        });
-      });
+        count = await collection.find({}, { skip: 5 }).count();
+        test.equal(5, count);
+
+        count = await db.collection('acollectionthatdoesn').count();
+        test.equal(0, count);
```

Comment:
suggest just removing this.  this made sense in the generic "count" test above, but here it makes no sense, because the test is testing `shouldCorrectlyExecuteCursorCountWithDottedCollectionName`